src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs

```diff
@@ -328,19 +330,81 @@ private void InsertEntry(NodeSearchElementViewModel entry, IEnumerable<string> c
+                    {
+                        targetClass = c as ClassesNodeCategoryViewModel;
+                        // As soon as ClassesNodeCategoryViewModel is found we should search 
+                        // through all it classes and save result.
+                        targetClassSuccessor = c.SubCategories.FirstOrDefault(c2 => c2.Name == next);
+                        return targetClassSuccessor != null;
+                    }
+
+                    return c.Name == next;
+                });
                 if (newTarget == default(NodeCategoryViewModel))
                 {
                     newTarget = isRoot ? new RootNodeCategoryViewModel(next) : new NodeCategoryViewModel(next);
+                    newTarget.FullCategoryName = path;
+                    // Situation when we to add only one new category and item as it child.
+                    // New category should be added to existing ClassesNodeCategoryViewModel.
+                    // Make notice: ClassesNodeCategoryViewModel is always first item in 
+                    // all subcategories.
+                    if (nameStack.Count == 0 && target.SubCategories.Count > 0 &&
+                        target.SubCategories[0] is ClassesNodeCategoryViewModel)
+                    {
+                        target.SubCategories[0].SubCategories.Add(newTarget);
+                        newTarget.Entries.Add(entry);
+                        return;
+                    }
+
+                    // We are here when target is the class. New category should be added
+                    // as child of it. So class will turn into usual category.
+                    // Here we are take class, remove it from ClassesNodeCategoryViewModel
+                    // and attach to it parrent.
+                    if (targetClass != null)
+                    {
+                        targetClass.SubCategories.Remove(target);
+                        targetClass.Parent.SubCategories.Add(target);
+                        if (targetClass.SubCategories.Count == 0)
+                            targetClass.Parent.SubCategories.RemoveAt(0);
+
+                        targetClass.Dispose();
+
+                        // Situation when we need to add only one new category and item.
+                        // Before adding of it we need create new ClassesNodeCategoryViewModel
+                        // as soon as new category will be a class.
+                        if (nameStack.Count == 0)
+                        {
+                            targetClass =
+                                new ClassesNodeCategoryViewModel(Configurations.ClassesDefaultName, target);
+
+                            target.SubCategories.Add(targetClass);
+                            target = targetClass;
+                        }
+                    }
                     target.SubCategories.Add(newTarget);
                     PlaceInNewCategory(entry, newTarget, nameStack);
```

Comment:
For the above one line:

```csharp
// Proceed to insert the new entry under 'newTarget' category with the remaining 
// name stack. In the first iteration this would have been 'MyNamespace.MyClass'.
PlaceInNewCategory(entry, newTarget, nameStack);
```